### DISCUSSION TITLE
T-Echo touch button not working

### DISCUSSION THREAD

**Discussion by mardzie** [meshtastic/firmware#5473] [Troubleshooting]
Hi, I recently bought a Lilygo T-Echo and updated it to the newest version of the meshtastic firmware  (2.5.11). Since then the touch button at the top does nothing. 

I have tried to downgrade the firmware to 2.5.9 and upgrade it to the newest alpha 2.5.14 but it didn't show any effect.

Thank you in advance.

**Comment by GUVWAF** [comment]
See: https://meshtastic.org/docs/hardware/devices/lilygo/techo/buttons/

The touch button now only updates the screen (e.g. show a new timestamp) or wakes it up from the screensaver. Only Button 2 cycles through the screens.

**Reply by mardzie** [reply]
Thank you for your reply but I can not see it updating the screen.

**Reply by GUVWAF** [reply]
It depends on what is displayed on the screen, but if nothing changed since it showed this screen, you won't see a difference.

**Reply by mardzie** [reply]
Isn't there a fast change to black then white again black again and in the end white? It is an e-ink display. As far as I know that fast change is a full update frome one.

**Reply by GUVWAF** [reply]
Several things (e.g. number of pixels changed, number of updates since last full refresh, etc.) affect whether it will do a full refresh or partial refresh.
